### ISSUE TITLE
Support `path` specification for git sources

### ISSUE THREAD

**Issue by cwjohnston** [hw-labs/batali#52]
librarian-chef supports a `path` specification for git sources, so that we can grab a cookbook from a subdirectory of the target repository, e.g.:

```
cookbook 'zncrypt', :git => 'git://github.com/cwjohnston/chef-zncrypt.git', :ref => '9d713d67c08735d26759d7af7a697d5d8638ce42', :path => 'zncrypt'
```

Currently batali seems to trip on this, treating the source as a local path:

```
$ bundle exec batali resolve --debug
[Batali]: Loading sources... [DEBUG]: Cache directory to persist cookbooks: /Users/REDACTED/.batali/cache
error!
[ERROR]: Reason - No such file or directory - Failed to locate metadata file in cookbook directory! (path: zncrypt)
ERROR: Errno::ENOENT: No such file or directory - Failed to locate metadata file in cookbook directory! (path: zncrypt)
```

**Comment by chrisroberts** [comment]
Need to make source check a bit less lazy for this (likely just an update to the ordering keys are checked) and then apply optional path on install copy from unpacked asset.

**Comment by chrisroberts** [comment]
Released in 0.2.26
